test/unit/reductions/exponentiated_gradient/test_exponentiatedgradient_smoke.py

```diff
@@ -238,3 +241,36 @@ def test_simple_fit_predict(self, Constraints):
         expgrad.fit(pd.DataFrame(X1), pd.Series(labels),
                     sensitive_features=pd.Series(sensitive_features))
         expgrad.predict(pd.DataFrame(X1))
+
+    def test_constant_predictor(self):
+        # Setup with data designed to result in "all single class"
+        # at some point in the grid
+        array_size = 1000
+        X_dict = {
+            "c": np.random.random(array_size),
+            "d": np.random.random(array_size)
+        }
+        X = pd.DataFrame(X_dict)
+
+        y = np.zeros(array_size)
+        y[0] = 1
+        A = np.zeros(array_size)
+        A = A + 11
+        A[0] = 10
+
+        estimator = LogisticRegression(solver='liblinear',
+                                       fit_intercept=True,
+                                       random_state=97)
+        expgrad = ExponentiatedGradient(estimator, DemographicParity())
+
+        # Following line should not throw an exception
+        expgrad.fit(X, y, sensitive_features=A)
```

Comment:
This `fit()` call is not triggering the single-class case. @MiroDudik , is there an easy way of forcing the algorithm to produce a single label? The approach above is borrowed from the equivalent `GridSearch` test, but expgrad is avoiding the problem.

Reply:
Could you just provide only one class?

Reply:
Actually, now I look, yes. I thought we had an upfront check that required a binary input, but it actually only requires that the unique values of Y are a _subset_ of {0, 1}.

Reply:
We could restrict it further, but I don't see why...

Reply:
I'm happy with not adding more restrictions :-)